File: cogs/5000.py
import discord
from discord.ext import commands
from discord import app_commands
import urllib.parse


class gosentyouen(commands.Cog):
    def __init__(self, bot: commands.Bot) -> None:
        self.bot: commands.Bot = bot

    @app_commands.choices(
        type=[
            app_commands.Choice(name="png(お勧め)", value="png"),
            app_commands.Choice(name="jpg", value="jpg"),
            app_commands.Choice(name="webp", value="webp"),
        ]
    )
    @app_commands.choices(
        hoshii=[
            app_commands.Choice(name="固定する", value="true"),
            app_commands.Choice(name="固定しない", value="false"),
        ]
    )
    @app_commands.choices(
        quality=[
            app_commands.Choice(name="低", value="30"),
            app_commands.Choice(name="中", value="70"),
            app_commands.Choice(name="高", value="100"),
        ]
    )
    @app_commands.choices(
        noalpha=[
            app_commands.Choice(name="白にする", value="true"),
            app_commands.Choice(name="白にしない", value="false"),
        ]
    )
    @app_commands.choices(
        rainbow=[
            app_commands.Choice(name="虹色にする", value="true"),
            app_commands.Choice(name="虹色にしない", value="false"),
        ]
    )
    @app_commands.describe(
        hoshii="下部文字列を「欲しい！」に固定する",
        noalpha="背景色を白にする",
        rainbow="虹色にする",
        type="画像拡張子",
        quality="画質(1-100)",
        bottom="下部文字列",
        top="上部文字列",
    )
    @app_commands.command(name="5000", description="5000兆円ほしいを生成します。")
    async def gosen(
        self,
        i: discord.Interaction,
        top: str,
        bottom: str,
        type: str,
        quality: str,
        hoshii: str,
        noalpha: str,
        rainbow: str,
    ):
        base_url = f"https://gsapi.cbrx.io/image?top={top}&bottom={bottom}"
        url = (
            base_url
            + f"&type={type}&q={quality}"
            + f"&hoshii={hoshii}&noalpha={noalpha}&rainbow={rainbow}"
        )
        parse_url = urllib.parse.quote(url)
        embed = discord.Embed(title="5000兆円ほしい!!", color=0x3498DB)
        embed.set_image(url=parse_url)
        embed.set_footer(text="生成: 5000choyen-api")
        await i.response.send_message(embed=embed)


async def setup(bot: commands.Bot) -> None:
    await bot.add_cog(gosentyouen(bot))

File: commands/individual/5000.ts
import { isMessageInstance } from "@sapphire/discord.js-utilities";
import { Command, ChatInputCommand } from "@sapphire/framework";
import { MessageEmbed } from "discord.js";
export class PingCommand extends Command {
  public constructor(context: Command.Context, options: Command.Options) {
    super(context, { ...options });
  }

  public override registerApplicationCommands(
    registry: ChatInputCommand.Registry
  ) {
    registry.registerChatInputCommand((builder) =>
      builder
        .setName("5000")
        .setDescription("5000兆円ほしいを生成します。")
        .addStringOption((option) =>
          option.setName("top").setDescription("上部文字列").setRequired(true)
        )
        .addStringOption((option) =>
          option
            .setName("bottom")
            .setDescription("下部文字列")
            .setRequired(true)
        )
        .addStringOption((option) =>
          option
            .setName("type")
            .setDescription("画像の拡張子")
            .setChoices(
              { name: "png(お勧め)", value: "png" },
              { name: "jpg", value: "jpg" },
              { name: "webp", value: "webp" }
            )
            .setRequired(true)
        )
        .addStringOption((option) =>
          option
            .setName("quality")
            .setDescription("画像の画質")
            .setChoices(
              { name: "低", value: "30" },
              { name: "中", value: "70" },
              { name: "高", value: "100" }
            )
            .setRequired(true)
        )
        .addStringOption((option) =>
          option
            .setName("hoshii")
            .setDescription("下部文字列を「欲しい！」に固定する")
            .setChoices(
              { name: "固定する", value: "true" },
              { name: "固定しない", value: "false" }
            )
            .setRequired(true)
        )
        .addStringOption((option) =>
          option
            .setName("noalpha")
            .setDescription("背景色を白にする")
            .setChoices(
              { name: "白にする", value: "true" },
              { name: "白にしない", value: "false" }
            )
            .setRequired(true)
        )
        .addStringOption((option) =>
          option
            .setName("rainbow")
            .setDescription("虹色にする")
            .setChoices(
              { name: "虹色にする", value: "true" },
              { name: "虹色にしない", value: "false" }
            )
            .setRequired(true)
        )
    );
  }

  public async chatInputRun(interaction: Command.ChatInputInteraction) {
    const op = interaction.options;
    await interaction.reply({
      embeds: [
        new MessageEmbed()
          .setImage(
            encodeURI(
              `https://gsapi.cbrx.io/image?top=${op.getString(
                "top"
              )}&bottom=${op.getString("bottom")}&type=${op.getString(
                "type"
              )}&quality=${op.getString("quality")}&hoshii=${op.getString(
                "hoshii"
              )}&noalpha=${op.getString("noalpha")}&rainbow=${op.getString(
                "rainbow"
              )}`
            )
          )
          .setTitle("5000兆円ほしい!!")
          .setFooter({ text: "5000choyen-apiで生成しています。" })
          .setColor(0x3498db),
      ],
    });
  }
}
